__tests__/app.test.js

```diff
@@ -0,0 +1,56 @@
+const request = require('supertest');
+const {app, server} = require('../app.js');
+const db = require('../db/connection.js');
+const seed = require('../db/seeds/seed.js');
+const data = require('../db/data/test-data/index.js')
+
+beforeEach(() => {
+    return seed(data);
+});
+
+afterAll(() => {
+    return db.end().then(() => {return server.close()});
+})
+
+describe('Invalid endpoint requests', () => {
+    test('404: should return not found in response to get request to non-existent endpoint', () => {
+        return request(app)
+            .get('/api/does-not-exist')
+            .expect(404)
+    });    
+});
+describe('GET /api/topics', () => {
+    test('200: should return an array', () => {
+        return request(app)
+            .get('/api/topics')
+            .expect(200)
+            .then(({body}) => {
+                expect(Array.isArray(body.topics)).toBe(true);
+            });
+    });
+    test('200: should return expected number of topics', () => {
+        return request(app)
+            .get('/api/topics')
+            .expect(200)
+            .then(({body}) => {
+                expect(body.topics.length).toBe(data.topicData.length);
+            });
+    });
+    test('200: should return expected fields', () => {
+        return request(app)
+            .get('/api/topics')
+            .expect(200)
+            .then(({body}) => {
+                expect(Object.keys(body.topics[0])).toContain('slug', 'description');
+            });
+    });
+    test('200: should return expected field types', () => {
+        return request(app)
+            .get('/api/topics')
+            .expect(200)
+            .then(({body}) => {
+                expect(typeof body.topics[0].slug).toBe('string');
+                expect(typeof body.topics[0].description).toBe('string');
+            });
```

Comment:
Good tests.  Look into expect.objectContaining. It'll make your test suits looks tidy